src/Microsoft.ML.Core/Data/ReadOnlyMemoryUtils.cs

```diff
@@ -0,0 +1,464 @@
+﻿using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.ML.Runtime.Data
+{
+    public static class ReadOnlyMemoryUtils
+    {
+
+        /// <summary>
+        /// This method retrieves the raw buffer information. The only characters that should be
+        /// referenced in the returned string are those between the returned min and lim indices.
+        /// If this is an NA value, the min will be zero and the lim will be -1. For either an
+        /// empty or NA value, the returned string may be null.
+        /// </summary>
+        public static string GetRawUnderlyingBufferInfo(out int ichMin, out int ichLim, ReadOnlyMemory<char> memory)
+        {
+            MemoryMarshal.TryGetString(memory, out string outerBuffer, out ichMin, out int length);
```

Comment:
My attitude towards this is either we support `ReadOnlyMemory<char>` or we don't. If we do, then our methods on top of it should work, even if this fails. Fortunately it seems like this was only introduced because we wanted to just have a light "shim" on top of existing `DvText` methods, which is not really something we want to do.